lib/ansible/executor/module_common.py

```diff
@@ -740,6 +749,8 @@ def modify_module(module_name, module_path, module_args, task_vars=dict(), modul
 
             if os.path.basename(interpreter).startswith(b'python'):
                 lines.insert(1, to_bytes(ENCODING_STRING))
+        elif _check_bom(lines[0]):
+            raise AnsibleError("BOM detected in module (%s). Module scripts encoded with BOM are not supported." % module_name)
```

Comment:
for users, a nicer message that explains what BOM is might be nicer:

'Unsupported byte order mark detected, this type of file encoding is not supported for modules'